### PR TITLE
Stacktrace recovery message-related improvements

### PR DESCRIPTION
**Pull request by qwwdfsad**

### CODE REVIEW

**Comment by qwwdfsad** [comment]
I'm investigating Space mirror failures separately, it shouldn't prevent this PR from review

File: docs/topics/debugging.md [hunk 1]
```diff
@@ -63,7 +63,9 @@ Exception copy logic is straightforward:
   1) If the exception class implements [CopyableThrowable], [CopyableThrowable.createCopy] is used.
      `null` can be returned from `createCopy` to opt-out specific exception from being recovered.
   2) If the exception class has class-specific fields not inherited from Throwable, the exception is not copied.
-  3) Otherwise, one of the public exception's constructor is invoked reflectively with an optional `initCause` call.  
+  3) Otherwise, one of the public exception's constructor is invoked reflectively with an optional `initCause` call. 
+  4) If the reflective copy has a changed message (exception class modifies 'message' parameter before passing it to the superclass), 
+     the exception is not copied in order to preserve a human-readable message. [CopyableThrowable] does not have such limitation.
```
Comment (dkhalanskyjb) [inline docs/topics/debugging.md:68]:
```suggestion
     the exception is not copied in order to preserve a human-readable message. [CopyableThrowable] does not have such a limitation
     and allows the copy to have a `message` different from that of the original.
```

File: kotlinx-coroutines-core/jvm/test/exceptions/StackTraceRecoveryCustomExceptionsTest.kt [hunk 2]
```diff
@@ -71,4 +72,57 @@ class StackTraceRecoveryCustomExceptionsTest : TestBase() {
...
+            }
+        }
+        val ex = result.exceptionOrNull() ?: error("Expected to fail")
+        assertTrue(ex is WrongMessageException)
+        assertEquals("Token OK", ex.message)
+    }
+
+    @Test
+    fun testWrongMessageExceptionInChannel() = runTest {
+        // Separate code path
```
Comment (dkhalanskyjb) [inline kotlinx-coroutines-core/jvm/test/exceptions/StackTraceRecoveryCustomExceptionsTest.kt:92]: I don't get this comment.

**Review by dkhalanskyjb — COMMENTED** [review]
The change looks good, the only gripes from me regard the documentation.

**Pull request merged**